src/OSFramework/DataGrid/Grid/AbstractGrid.ts

```diff
@@ -149,10 +149,19 @@ namespace OSFramework.DataGrid.Grid {
                 (col) => col.config.binding
             );
             return Object.keys(metadata).some((source) => {
-                const newColumns = Object.keys(metadata[source]);
-                hasColumns = newColumns.every((column) => {
-                    return columns.indexOf(`${source}.${column}`) !== -1;
-                });
+                const isObject = typeof metadata[source] === 'object';
+                // If it is an object, let's iterate into the keys to check if there are new columns
+                if (isObject) {
+                    const newColumns = Object.keys(metadata[source]);
+                    hasColumns = newColumns.every((column) => {
```

Comment:
Use `some` instead of `every` in the inner loop because we only need to find one new column to return true and terminate the iteration early, which makes it more efficient.